beacon_node/eth2-libp2p/src/behaviour/handler/mod.rs

```diff
@@ -116,26 +140,40 @@ impl<TSpec: EthSpec> ProtocolsHandler for BehaviourHandler<TSpec> {
             Self::Error,
         >,
     > {
+        // TODO: is there any better place to check this?
+        if let KeepAlive::No = self.delegate.rpc().connection_keep_alive() {
+            if let KeepAlive::Yes = self.keep_alive {
+                // RPC inactivity timeout has passed
+                self.keep_alive = KeepAlive::Until(Instant::now() + self.shutdown_grace_time);
```

Comment:
I was thinking of having a timeout before we ungracefully shutdown. i.e: If we are trying to shutdown and the RPC (or any other sub-handler) still says KeepAlive::Yes.. we close anyway. This would ensure we get a disconnect if there is an error in the sub-handler that keeps it alive for some reason. 

We can use the RPC inactivity timeout to do this somewhat gracefully, i.e allow a goodbye etc.. but if it's not closed after a larger period of time the main handler closes regardless. By sending a `ProctocolsHandlerEvent::Close`